### PR TITLE
Use new `ignore_dynamic_beyond_limit` setting in logs and metrics data streams

### PR DESCRIPTION
**Pull request by felixbarny**
This reduces the risk of document loss if too many fields are added.

As these component templates are imported by Fleet, this also affects integrations.

### CODE REVIEW

**Review by axw — APPROVED** [review]

**Review by tommyers-elastic — APPROVED** [review]
LGTM

**Review by kpollich — APPROVED** [review]
LGTM! No concerns with this on the integrations side.

File: modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/LogsDataStreamIT.java [hunk 1]
```diff
@@ -716,6 +716,52 @@ public void testDefaultFieldCustomization() throws Exception {
...
+        List<Object> results = searchDocs(client, dataStreamName, """
+            {
+              "query": {
+                "match_all": { }
+              },
+              "fields": ["*"]
+            }
+            """);
+        assertEquals(1, results.size());
+        assertFalse(((List<String>) ((Map<String, ?>) results.get(0)).get("_ignored")).isEmpty());
```
Comment (dakrone) [inline modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/LogsDataStreamIT.java:761]: Are we able to assert which one was ignored?

**Review by dakrone — APPROVED** [review]
LGTM, I left ~one~ two comments

**Comment by dakrone** [comment]
Oh, we also need to bump the REGISTRY_VERSION for these templates, so that they get installed.

**Comment by felixbarny** [comment]
We'll need to remove the `ignore_malformed` part as it leads to issues: https://github.com/elastic/elasticsearch/pull/105324

**Pull request merged**